Feature enum my inject

rmauritsson opened the pull request.

Ispirett commented on main.rb:
```diff
@@ -0,0 +1,90 @@
+        end
+    end
+
+    def my_inject
```
my, inject should be able to take an argument.

rmauritsson replied:
Fixed

Ispirett commented on main.rb:
```diff
@@ -0,0 +1,90 @@
+        false
+    end
+
+    def my_none?
```
**my_count**  should be able to take an argument.

rmauritsson replied:
Done

Ispirett commented on main.rb:
```diff
@@ -0,0 +1,90 @@
+        true
+    end
+
+    def my_any?
```
**my_any?** should be able to take an argument.
your method should also have a guard case that handles and error when **no block is given**

rmauritsson replied:
Done

Ispirett commented on main.rb:
```diff
@@ -0,0 +1,90 @@
+        result
+    end
+
+    def my_all?
```
**my_all?** should be able to take an argument.
**your method should also have a guard case that handles and error when no block is given**
 undefined method `eturnr'

Ispirett commented on main.rb:
```diff
@@ -0,0 +1,90 @@
+        end
+    end
+
+    def my_select
```
your method should also have a guard case that handles and error when no block is given

Ispirett commented on main.rb:
```diff
@@ -0,0 +1,90 @@
+        self
+    end
+
+    def my_each_with_index
```
your method should also have a guard case that handles and error when no block is given

Ispirett commented on main.rb:
```diff
@@ -0,0 +1,90 @@
+module Enumerable
+    def my_each
```
your method should also have a guard case that handles and error when no block is given

rmauritsson commented:
Completed all the changes suggested

rmauritsson commented:
Modified the my_all? , my_any? and my_none? to respond when you pass a class as an argument and when you pass a regular expression.

Updated the README so that it's more descriptive.
Couldn't change the name of my pull request since am using the same PR but that is something I shall start doing for future projects

maelfosso commented on main.rb:
```diff
@@ -0,0 +1,157 @@
+
+    def my_all?(value = nil)
+        begin
+            if value == nil && !block_given?
```
```ruby
irb(main):013:0> array
=> [0, 8, 0, 6, 4, 4, 3, 5, 7, 2, 8, 7, 4, 6, 2, 3, 0, 0, 4, 8, 1, 7, 8, 2, 8, 5, 5, 8, 4, 0, 6, 7, 4, 3, 1, 8, 5, 5, 6, 3, 3, 0, 0, 0, 5, 8, 0, 1, 3, 2, 4, 5, 4, 4, 0, 5, 0, 0, 5, 2, 7, 6, 0, 5, 4, 7, 2, 2, 7, 6, 8, 6, 7, 5, 0, 6, 7, 4, 6, 5, 4, 7, 1, 6, 0, 3, 0, 7, 8, 3, 6, 0, 7, 2, 1, 5, 7, 4, 4, 0]
irb(main):014:0> array.all?(Integer)
=> true
irb(main):015:0> array.my_all?(Integer)
=> false
irb(main):016:0> 
```

maelfosso replied:
```ruby
irb(main):018:0> words = %w[dog door rod blade]
=> ["dog", "door", "rod", "blade"]
irb(main):019:0> words.all?(/d/)
=> true
irb(main):020:0> words.my_all?(/d/)
=> false
irb(main):021:0> 
```

maelfosso commented on main.rb:
```diff
@@ -0,0 +1,180 @@
+        rescue => e
+            puts "Exception Class: #{e.class.name}"
+      end
+    end
```
```ruby
irb(main):023:0> array
=> [0, 8, 0, 6, 4, 4, 3, 5, 7, 2, 8, 7, 4, 6, 2, 3, 0, 0, 4, 8, 1, 7, 8, 2, 8, 5, 5, 8, 4, 0, 6, 7, 4, 3, 1, 8, 5, 5, 6, 3, 3, 0, 0, 0, 5, 8, 0, 1, 3, 2, 4, 5, 4, 4, 0, 5, 0, 0, 5, 2, 7, 6, 0, 5, 4, 7, 2, 2, 7, 6, 8, 6, 7, 5, 0, 6, 7, 4, 6, 5, 4, 7, 1, 6, 0, 3, 0, 7, 8, 3, 6, 0, 7, 2, 1, 5, 7, 4, 4, 0]
irb(main):024:0> array.any?(Integer)
=> true
irb(main):025:0> array.my_any?(Integer)
=> false
irb(main):026:0> 
```

maelfosso replied:
```ruby
irb(main):028:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):029:0> words.any?(/d/)
=> true
irb(main):030:0> words.my_any?(/d/)
=> false
irb(main):031:0>
```

rmauritsson replied:
Modified and tested them they should be working now.

rmauritsson commented:
[my_any tests](https://drive.google.com/file/d/1eii2movCbF-mCQJl3Z8K0DzUMwVqg3JW/view?usp=sharing)

Modified the method and tested it as shown in the screenshot
